kroma-validator/challenger.go

```diff
@@ -598,25 +619,66 @@ func (c *Challenger) submitChallengeTx(tx *types.Transaction) error {
 	return c.cfg.TxManager.SendTransaction(c.ctx, tx).Err
 }
 
-// HasEnoughDeposit checks if challenger has enough deposit to bond when creating challenge.
-func (c *Challenger) HasEnoughDeposit(ctx context.Context) (bool, error) {
+// CanCreateChallenge checks if challenger is in the status that can make challenge.
+func (c *Challenger) CanCreateChallenge(ctx context.Context, outputIndex *big.Int) (bool, error) {
 	cCtx, cCancel := context.WithTimeout(ctx, c.cfg.NetworkTimeout)
 	defer cCancel()
-	balance, err := c.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), c.cfg.TxManager.From())
-	if err != nil {
-		return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
-	}
-	c.metr.RecordDepositAmount(balance)
+	from := c.cfg.TxManager.From()
+	if c.IsValPoolTerminated(outputIndex) {
+		validatorStatus, err := c.valManagerContract.GetStatus(optsutils.NewSimpleCallOpts(cCtx), from)
+		if err != nil {
+			return false, fmt.Errorf("failed to fetch the validator status: %w", err)
+		}
+
+		if isInJail, err := c.IsInJail(ctx); err != nil {
+			return false, err
+		} else if isInJail {
+			c.log.Warn("validator is in jail")
+			return false, nil
+		}
+
+		if validatorStatus != StatusActive {
+			c.log.Warn("validator is not in the status that can create a challenge", "status", validatorStatus)
+			return false, nil
+		}
+	} else {
+		balance, err := c.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), from)
+		if err != nil {
+			return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
+		}
+		c.metr.RecordDepositAmount(balance)
+
+		if balance.Cmp(c.requiredBondAmount) == -1 {
+			c.log.Warn(
+				"deposit is less than bond attempt amount",
+				"requiredBondAmount", c.requiredBondAmount,
+				"deposit", balance,
+			)
+			return false, nil
+		}
 
-	if balance.Cmp(c.requiredBondAmount) == -1 {
-		c.log.Warn("deposit is less than bond amount", "required", c.requiredBondAmount, "deposit", balance)
-		return false, nil
+		c.log.Info("deposit amount and bond amount", "deposit", balance, "bond", c.requiredBondAmount)
 	}
-	c.log.Info("deposit amount and bond amount", "deposit", balance, "bond", c.requiredBondAmount)
 
 	return true, nil
 }
 
+func (c *Challenger) IsValPoolTerminated(outputIndex *big.Int) bool {
```

Comment:
Let's change in the future. I think like `isV2` is more Intuitive, but replace V2 to the name of the upgrade for validator system v2 :)